pkg/controller/worker/machines.go

```diff
@@ -168,6 +168,22 @@ func (w *workerDelegate) generateMachineConfig() error {
 				"blockDevices": blockDevices,
 			}
 
+			if workerConfig.NodeTemplate != nil {
+				machineClassSpec["nodeTemplate"] = machinev1alpha1.NodeTemplate{
+					Capacity:     workerConfig.NodeTemplate.Capacity,
+					InstanceType: pool.MachineType,
+					Region:       w.worker.Spec.Region,
+					Zone:         zone,
+				}
+			} else if pool.NodeTemplate != nil {
+				machineClassSpec["nodeTemplate"] = machinev1alpha1.NodeTemplate{
+					Capacity:     pool.NodeTemplate.Capacity,
+					InstanceType: pool.MachineType,
+					Region:       w.worker.Spec.Region,
+					Zone:         zone,
+				}
+			}
```

Comment:
Just to confirm: I assume we give the configuration in the providerConfig priority over the pool config due to the scenarios where the user provide the values on its own e.g. scaleMP?

Reply:
That's right, @dkistner.  The configuration in the `providerConfig` is prioritised over the pool config due to the scenarios where the user provide the values on its own e.g. scaleMP

Reply:
Yes you are correct.